src/main/java/com/aws/greengrass/ipc/SubscribeToCertificateUpdatesOperationHandler.java

```diff
@@ -0,0 +1,167 @@
+                throw new UnauthorizedError(e.getMessage());
+            }
+            try {
+                String certificateType = getCertificateTypeFromCertificateOptions(
+                        subscribeToCertificateUpdatesRequest.getCertificateOptions());
+                if (CertificateType.SERVER.getValue().equals(certificateType)) {
+                    final String csr = getCSRForServer();
+                    Consumer<X509Certificate> serverCertificateCallback = this::forwardServerCertificates;
+                    certificateManager.subscribeToServerCertificateUpdates(csr, serverCertificateCallback);
+                }
+            } catch (NoSuchAlgorithmException | IOException | OperatorCreationException
+                    | KeyStoreException | CsrProcessingException e) {
+                logger.atError().cause(e).log("Unable to create the certificate signing request");
+                throw new ServiceError(String.format("Subscribe to certificate update failed with error %s", e));
+            }
+            return new SubscribeToCertificateUpdatesResponse();
+        });
+    }
+
+
+    private String getCertificateTypeFromCertificateOptions(CertificateOptions certificateOptions) {
+        validateCertificateOptions(certificateOptions);
+        CertificateType certificateType = certificateOptions.getCertificateType();
+        if (certificateType == null) {
+            logger.atError().kv(COMPONENT_NAME, serviceName).log(NO_CERT_TYPE_ERROR);
+            throw new InvalidArgumentsError(NO_CERT_TYPE_ERROR);
+        }
+        return certificateType.getValue();
+    }
+
+    private void validateCertificateOptions(CertificateOptions certificateOptions) {
+        if (certificateOptions == null) {
+            logger.atError().kv(COMPONENT_NAME, serviceName).log(NO_CERT_OPTIONS_ERROR);
+            throw new InvalidArgumentsError(NO_CERT_OPTIONS_ERROR);
+        }
+    }
+
+    private void forwardServerCertificates(X509Certificate certificate) {
+        CertificateUpdate certificateUpdate = new CertificateUpdate();
+        try {
+            certificateUpdate
+                    .withCertificate(EncryptionUtils.encodeToPem(PEM_BOUNDARY_CERTIFICATE, certificate.getEncoded()))
+                    .withCaCertificates(this.certificateManager.getCACertificates())
+                    .withPublicKey(
+                            EncryptionUtils.encodeToPem(PEM_BOUNDARY_PUBLIC_KEY, keyPair.getPublic().getEncoded()))
+                    .withPrivateKey(
+                            EncryptionUtils.encodeToPem(PEM_BOUNDARY_PRIVATE_KEY, keyPair.getPrivate().getEncoded()));
+        } catch (CertificateEncodingException | IOException | KeyStoreException e) {
+            logger.atError().cause(e).log("Unable to attach certificates to the response");
+            throw new ServiceError(String.format("Subscribe to certificate update failed with error %s", e));
```

Comment:
```suggestion
            throw new ServiceError("Subscribe to certificate update failed. Check Greengrass log for details.");
```

Reply:
Updating now. I was waiting for Joe's thoughts because you mentioned him.  :)